### ISSUE TITLE
What is the recommended way to test an Android application in a multi-language environment?

### ISSUE THREAD

**Issue by tswast** [travis-ci/travis-ci#5779]
I cannot use the Android language environment because our repository also holds Go, PHP, Python, Java, and iOS apps (and cannot be split at this time).

For installing Go, you provide [gimme](https://github.com/travis-ci/gimme). Is there a similar script for installing the Android development environment? I've found it difficult to keep up when trying to download the SDK myself.

**Comment by BanzaiMan** [comment]
## TL; DR

You can't do everything you are asking, but this config gets you far:

```yaml
sudo: required
dist: trusy
group: edge
```

## Details

Android builds require SDKs that require a lot of disk space, and there is no way to manage them.

iOS builds are available on the Mac. See issues listed in https://github.com/travis-ci/travis-ci/issues/2320 for more details. Due to disk space concerns, Mac image is unlikely to get much more stuff pre-installed than it has now.

To complicate things, other languages (PHP, Python) have similar issues where tools are available on the language's build image, and that image only.

Having said all that, you can use Trusty's [`mega` image](https://docs.travis-ci.com/user/trusty-ci-environment#Image-differences-from-Precise) to achieve some of the things you listed (I suggest using `group: edge` at the top level of `.travis.yml` to get more recent but experimental build image).
